va/va_dec_av1.h

```diff
@@ -0,0 +1,546 @@
+            uint8_t         reserved                                    : 3;
+        } bits;
+        uint8_t             value;
+    } loop_filter_info_fields;
+
+    /** \brief The adjustment needed for the filter level based on
+     *  the chosen reference frame.
+     *  value range [-63..63].
+     */
+    int8_t                  ref_deltas[8];
+
+    /** \brief The adjustment needed for the filter level based on
+     *  the chosen mode.
+     *  value range [-63..63].
+     */
+    int8_t                  mode_deltas[2];
+
+    /** \brief quantization
+     */
+    /** \brief Y AC index
+     *  value range [0..255]
+     */
+    uint16_t                base_qindex;
+    /** \brief Y DC delta from Y AC
+     *  value range [-63..63]
+     */
+    int8_t                  y_dc_delta_q;
+    /** \brief U DC delta from Y AC
+     *  value range [-63..63]
+     */
+    int8_t                  u_dc_delta_q;
+    /** \brief U AC delta from Y AC
+     *  value range [-63..63]
+     */
+    int8_t                  u_ac_delta_q;
+    /** \brief V DC delta from Y AC
+     *  value range [-63..63]
+     */
+    int8_t                  v_dc_delta_q;
+    /** \brief V AC delta from Y AC
+     *  value range [-63..63]
+     */
+    int8_t                  v_ac_delta_q;
+
+    /** \brief quantization_matrix
+     */
+    union
+    {
+        struct
+        {
+            uint16_t        using_qmatrix                               : 1;
+            /** \brief qm level
+             *  value range [0..15]
+             *  Invalid if using_qmatrix equals 0.
+             */
+            uint16_t        qm_y                                        : 4;
+            uint16_t        qm_u                                        : 4;
+            uint16_t        qm_v                                        : 4;
+            uint16_t        reserved                                    : 3;
+        } bits;
+        uint16_t            value;
+    } qmatrix_fields;
+
+
+    union
+    {
+        struct
+        {
+            /** \brief delta_q parameters
+             */
+            uint32_t        delta_q_present_flag                        : 1;
+            uint32_t        log2_delta_q_res                            : 2;
+
+            /** \brief delta_lf parameters
+             */
+            uint32_t        delta_lf_present_flag                       : 1;
+            uint32_t        log2_delta_lf_res                           : 2;
+
+            /** \brief CONFIG_LOOPFILTER_LEVEL
+             */
+            uint32_t        delta_lf_multi                              : 1;
+
+            /** \brief read_tx_mode
+             *  value range [0..6]
+             */
+            uint32_t        tx_mode                                     : 2;
+
+            /** \brief read_frame_reference_mode
+             */
+            uint32_t        reference_mode                              : 2;

```

Comment:
it's no read_frame_refence_mode in "Version 1.0.0 with Errata 1"
could you help check?

Reply:
I checked aom reference code, and found read_frame_reference_mode is there. however in Spec, it is named as  frame_reference_mode.

Reply:
This makes me more confused.  
Is reference_mode equals to reference_select in frame_reference_mode?Apparently, it's not. because  reference_select  has only 1 bit, but reference_mode has 2 bits.
Is it the same thing as comp_mode?  not likely, because comp_mode only has two value SINGLE_REFERENCE, COMPOUND_REFERENCE. 1 bits is enough for them.

IMO, we'd better stick name and document to spec. aom just one implement for spec. the name may change after spec finalized. and it may add an unnecessary definition for optimization/code simplify purpose. 
If we stick to spec, the one who implement hw codec only need to check the spec. not the aom code.

Reply:
All right, we will stick name to spec, actually in aom code, reference mode has below types.
typedef enum {
  SINGLE_REFERENCE = 0,
  COMPOUND_REFERENCE = 1,
  REFERENCE_MODE_SELECT = 2,
  REFERENCE_MODES = 3,
} REFERENCE_MODE;
so we use 2 bits, however, we will swicth to spec defination <reference_select> and use 1 bit.

Reply:
The SINGLE_REFERENCE/COMPOUND_REFERENCE is for macroblock level(in read_ref_frames). But libva's user only handles stream in slice level(in frame_reference_mode). this why the caller can only give you a bool value(reference_select)